### ISSUE TITLE
File and stream assertion 

### ISSUE THREAD

**Issue by hellboy81** [fluentassertions/fluentassertions#961]
Is there possible to port [FileAndStreamAssertionsExamples](https://github.com/joel-costigliola/assertj-examples/blob/master/assertions-examples/src/test/java/org/assertj/examples/FileAndStreamAssertionsExamples.java) to FluentAssertion?

Assertion two files are (not) same with **diff feature** is important.

Diffs respecting xml structs are welcome

**Comment by dennisdoomen** [comment]
You can create your [own extension](https://fluentassertions.com/extensibility/) and propose a PR.

**Comment by hellboy81** [comment]
own extension is may be not required due on [XMlUnit](https://github.com/jovaneyck/XmlUnitDemo)

**Comment by jnyrup** [comment]
@hellboy81 We will not take a dependency on `xmlunit`.

As mentioned, you are welcome to either create your own extension which may be depend on `xmlunit` or propose a PR without the dependency.

If you're not comfortable with implementing an extension, please list what features you'd like to see.
E.g.
```c#
File file = new File("c:\foo\bar.txt");

file.Should().Exist();
file.Should().BeReadable();
file.Should().BeWriteable();
```
Then somebody else might see the usefulness in those assertions methods and contribute the implementations.

**Comment by jnyrup** [comment]
Closing for now.
Feel free to re-open, if you've got clarifications

**Comment by julealgon** [comment]
I just came across the need to validate whether a `Stream` result had its `CanWrite` property set to `false`. I was expecting the base FluentAssertions library to have some support for this type but there seems to be none at the moment.

After searching a bit I found this closed issue. Should I open a new issue requesting this @jnyrup ?

For now, I'll probably leave my code as 
```csharp
stream.CanWrite.Should().BeFalse();
```

Though I may end up creating a very small extension with just this method for now as something like `Should().BeWriteable()` as you suggested above for `FileInfo`, or even a variation like `Should().BeReadOnly` that checks for `CanWrite=false` _and_ `CanRead=true` at the same time which would be even more semantic in my case.
